admin-ui/plugins/auth-server/components/AuthN/LdapListingPage.js

```diff
@@ -0,0 +1,233 @@
+import React, { useState, useEffect, useContext, useCallback } from 'react'
+import MaterialTable from '@material-table/core'
+import { Paper } from '@mui/material'
+import { useDispatch, useSelector } from 'react-redux'
+import applicationStyle from 'Routes/Apps/Gluu/styles/applicationstyle'
+import { useTranslation } from 'react-i18next'
+import { useCedarling } from '@/cedarling'
+import { ThemeContext } from 'Context/theme/themeContext'
+import getThemeColor from 'Context/theme/config'
+import { getLdapList, setCurrentItem } from '../../redux/features/authNLdapSlice'
+import { useNavigate } from 'react-router-dom'
+import customColors from '@/customColors'
+import { LDAP_READ, LDAP_WRITE, LDAP_DELETE } from 'Utils/PermChecker'
+import { STRINGS } from '../../helper/constants'
+import GluuFormDetailRow from 'Routes/Apps/Gluu/GluuFormDetailRow'
+import { Container, Row, Col } from 'Components'
+import GluuViewWrapper from 'Routes/Apps/Gluu/GluuViewWrapper'
+import GluuDialog from 'Routes/Apps/Gluu/GluuDialog'
+import SetTitle from '@/utils/SetTitle'
+
+function LdapListingPage() {
+  const dispatch = useDispatch()
+  const { t } = useTranslation()
+  const { hasCedarPermission, authorize } = useCedarling()
+
+  const [limit] = useState(10)
+  const theme = useContext(ThemeContext)
+  const selectedTheme = theme.state.theme
+  const themeColors = getThemeColor(selectedTheme)
+  const backgroundStyle = { backgroundColor: '#f5f5f5', padding: '20px' }
+  const bgThemeColor = { background: themeColors.background }
+
+  const ldapListRaw = useSelector((state) => state.authNLdap.ldapList)
+  const loading = useSelector((state) => state.authNLdap.loading)
+  const [ldapList, setLdapList] = useState([])
+  const [myActions, setMyActions] = useState([])
+  const navigate = useNavigate()
+
+  const [modal, setModal] = useState(false)
+  const [selectedRow, setSelectedRow] = useState(null)
+  const toggle = useCallback(() => setModal(!modal), [modal])
+  const { permissions } = useSelector((state) => state.cedarPermissions)
+
+  SetTitle('LDAP Server')
+
+  function LdapDetailRow({ label, value }) {
+    return (
+      <Row>
+        <Col sm={6}>
+          <GluuFormDetailRow label={label[0]} value={value[0]} />
+        </Col>
+        {label[1] && value[1] && (
+          <Col sm={6}>
+            <GluuFormDetailRow label={label[1]} value={value[1]} />
+          </Col>
+        )}
+      </Row>
+    )
+  }
+
+  useEffect(() => {
+    const initPermissions = async () => {
+      const permissions = [LDAP_READ, LDAP_WRITE, LDAP_DELETE]
+      for (const permission of permissions) {
+        await authorize([permission])
+      }
+    }
+    initPermissions()
+    dispatch(getLdapList())
+  }, [dispatch])
```

Comment:
Authorize once with all permissions, add deps and error handling.

Current loop awaits each permission, misses authorize in deps, and lacks a try/catch.

```diff
-useEffect(() => {
-  const initPermissions = async () => {
-    const permissions = [LDAP_READ, LDAP_WRITE, LDAP_DELETE]
-    for (const permission of permissions) {
-      await authorize([permission])
-    }
-  }
-  initPermissions()
-  dispatch(getLdapList())
-}, [dispatch])
+useEffect(() => {
+  const init = async () => {
+    try {
+      await authorize([LDAP_READ, LDAP_WRITE, LDAP_DELETE])
+    } catch (e) {
+      console.error('LDAP permission authorize failed:', e)
+    }
+    dispatch(getLdapList())
+  }
+  init()
+}, [authorize, dispatch])
```
Also avoid shadowing “permissions” (see Lines 63-66).